components/board/numato/board.go

```diff
@@ -411,7 +428,7 @@ func connect(ctx context.Context, name resource.Name, conf *Config, logger golog
 	}
 
 	b.lines = make(chan string)
-	go b.readThread()
+	b.readThread()
```

Comment:
This totally works. Another way that keeps the code less indented is to make this part
```go
b.activeBackgroundWorkers.Add(1)
utils.ManagedGo(b.readThread, b.activeBackgroundWorkers.Done)
```
and then leave the implementation of `readThread` largely unchanged. I'm content with either approach, though I think this suggestion is easier to reason about because `readThread` itself can just focus on reading the thread, and the background workers are only mentioned in the constructor and destructor (`connect` and `Close`).